src/containers/other/NFTView/elements/Actions/Actions.tsx

```diff
@@ -68,83 +85,103 @@ const Actions = ({
 	isOwnedByUser,
 	wildPriceUsd,
 	refetch,
+	bidData,
+	isLoading,
+	domainMetadata,
+	domain,
+	setIsViewBidsOpen,
+	isViewBidsOpen,
+	setIsSetBuyNowOpen,
+	isSetBuyNowOpen,
 }: ActionsProps) => {
+	//- Condition helpers
+	const isBidData = bidData && bidData?.length > 0;
+	const isYourBids = !isOwnedByUser && Boolean(yourBid);
+	const isPlaceBid = !isOwnedByUser && isBiddable === true;
+	const isSetBuyNow = isOwnedByUser === true && Boolean(domainId);
+	const isBuyNow = Boolean(buyNowPrice) && !isOwnedByUser && Boolean(domainId);
+	const isViewBids =
+		isOwnedByUser !== undefined && isBiddable === true && Boolean(isBidData);
+
+	const highestBidTextValue =
+		Boolean(highestBid) && Boolean(wildPriceUsd)
+			? wrapFiat(highestBid! * wildPriceUsd!)
+			: LABELS.NO_BIDS_PLACED;
+
+	const yourBidTextValue =
+		yourBid && wildPriceUsd
+			? wrapFiat(Number(formatEther(yourBid.amount)) * wildPriceUsd)
+			: '';
+
+	const placeBidButtonTextValue =
+		!yourBid || (yourBid && Number(formatEther(yourBid.amount)) === highestBid)!
+			? LABELS.PLACE_A_BID
+			: LABELS.REBID;
+
+	const buyNowPriceValue =
+		buyNowPrice && wildPriceUsd
+			? wrapFiat(buyNowPrice * wildPriceUsd)
+			: LABELS.NO_BUY_NOW;
+
 	/**
 	 * This is a pretty messy structure - could be refactored moving forward
 	 */
 	const actions: { [action in ACTION_TYPES]: ActionBlock } = {
 		[ACTION_TYPES.BuyNow]: {
 			amount: buyNowPrice,
-			label: 'Buy Now (WILD)',
-			amountUsd:
-				buyNowPrice && wildPriceUsd
-					? wrapFiat(buyNowPrice * wildPriceUsd)
-					: 'No buy now set',
+			label: `${LABELS.BUY_NOW} (${CURRENCY.WILD})`,
+			amountUsd: buyNowPriceValue,
 			buttonComponent: (isTextButton?: boolean) => (
 				<BuyNowButton
 					onSuccess={refetch}
-					buttonText="Buy Now"
+					buttonText={LABELS.BUY_NOW}
 					domainId={domainId ?? ''}
 					isTextButton={isTextButton}
 					className={cx({ TextButton: isTextButton })}
 				/>
 			),
-			isVisible: Boolean(buyNowPrice) && !isOwnedByUser && Boolean(domainId),
+			isVisible: isBuyNow,
 			testId: TEST_ID.BUY_NOW,
 		},
 		[ACTION_TYPES.SetBuyNow]: {
 			amount: buyNowPrice ? buyNowPrice : '-',
-			label: 'Buy Now (WILD)',
-			amountUsd:
-				buyNowPrice && wildPriceUsd
-					? wrapFiat(buyNowPrice * wildPriceUsd)
-					: 'No buy now set',
+			label: `${LABELS.BUY_NOW} (${CURRENCY.WILD})`,
+			amountUsd: buyNowPriceValue,
 			buttonComponent: (isTextButton?: boolean) => (
 				<SetBuyNowButton
 					onSuccess={refetch}
-					buttonText={buyNowPrice ? 'Edit Buy Now' : 'Set Buy Now'}
+					buttonText={buyNowPrice ? LABELS.EDIT_BUY_NOW : LABELS.SET_BUY_NOW}
 					domainId={domainId ?? ''}
 					isTextButton={isTextButton}
 					className={cx({ TextButton: isTextButton })}
+					setIsSetBuyNowOpen={setIsSetBuyNowOpen}
+					isSetBuyNowOpen={isSetBuyNowOpen}
```

Comment:
Said above, but shouldn't need to pass these values in as modal rendering is controlled within the `SetBuyNowButton` component (`ViewBidsButton` should behave this way too)